### ISSUE TITLE
Slow speed

### ISSUE THREAD

**Issue by DX3906G** [zju3dv/DetectorFreeSfM#39]
Really a great work! But when I test DetectorFreeSfM on my own dataset (100 images of 640*480), I notice that it takes more than one hour to finish the coarse matching. In contrast, Colomap can finish it in 10 minutes. 
I would like to know the reason for this problem. Is there any way to optimize it? I notice that the 'img_pair_strategy' config of neuralsfm is set to 'exhaustive'. Is this related to the problem?

**Comment by hxy-123** [comment]
Yes, exhaustive matching is slow especially when performing matching on a large number of images, as it constructs C_n^2 pairs. Actually, we use exhaustive matching just for following other baselines in experiments for fair comparison. In practice, you can use sequential matching if image orders are known. Moreover, you can open parallel matching to speed up the matching phase.

**Comment by hxy-123** [comment]
One more thing, you can lower the image resize setting for high matching speed in your setting since your image size is 640*480 where the default image resizing param is 1200.

**Comment by DX3906G** [comment]
Besides, I meet another problem. In the 'colmap_coarse' result folder of my own dataset (100 images of 640*480, textureless object), the whole sequence is divided into serval subfolders, each contains a part of the sequence. What's the reason for this?
![image](https://github.com/zju3dv/DetectorFreeSfM/assets/74975608/57983ccb-188a-40ab-b2b0-469a48f34ba9)

**Comment by Bin-ze** [comment]
> Besides, I meet another problem. In the 'colmap_coarse' result folder of my own dataset (100 images of 640*480, textureless object), the whole sequence is divided into serval subfolders, each contains a part of the sequence. What's the reason for this? ![image](https://private-user-images.githubusercontent.com/74975608/338989545-57983ccb-188a-40ab-b2b0-469a48f34ba9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2MDMxMDgsIm5iZiI6MTcxODYwMjgwOCwicGF0aCI6Ii83NDk3NTYwOC8zMzg5ODk1NDUtNTc5ODNjY2ItMTg4YS00MGFiLWIyYjAtNDY5YTQ4ZjM0YmE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE3VDA1NDAwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExYmE0N2MyYWNjZWJjZDZjNDk5NTgxNmQ3MjE3ZDRkOWJiMzRjNTlmOTc3NjQwNmMwYTBhZDhmY2NiN2JlZTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Ia-Tu0XvTtGkqnLVJ9W2aKP4DvlWpHYsEgLc00CGNqc)

This is because there are too few matches between frames, so it is impossible to reconstruct a unified scene.
One solution is to use colmap's official alignment API to merge sub-scenes

**Comment by wwj-53** [comment]
@hxy-123 @DX3906G    
I set imd_pair_Strategy to sequential in dfsfm.yaml（ img_pair_strategy: sequential #exhaustive  sequential)， It will report an error:
(eval_core_ray_wrapper pid=888415) 2024-07-15 10:49:06.431 | ERROR    | __main__:eval_core_ray_wrapper:93 - Error captured:
(eval_core_ray_wrapper pid=888412) 2024-07-15 10:49:06.430 | INFO     | __main__:eval_core:36 - Worker: 1 will process: [], total: 0 scenes
Setting it to exhaustive will not result in errors, but it runs very, very slowly.
How should I make changes? Thank you!

**Comment by LaFeuilleMorte** [comment]
> @hxy-123 @DX3906G I set imd_pair_Strategy to sequential in dfsfm.yaml（ img_pair_strategy: sequential #exhaustive sequential)， It will report an error: (eval_core_ray_wrapper pid=888415) 2024-07-15 10:49:06.431 | ERROR | **main**:eval_core_ray_wrapper:93 - Error captured: (eval_core_ray_wrapper pid=888412) 2024-07-15 10:49:06.430 | INFO | **main**:eval_core:36 - Worker: 1 will process: [], total: 0 scenes Setting it to exhaustive will not result in errors, but it runs very, very slowly. How should I make changes? Thank you!

![image](https://github.com/user-attachments/assets/fa0db7a2-7f11-46c4-a8eb-d2c4d9ae6ef7)
YOU should use 'pair_from_index'
